drivers/spi/spi_sifive.c

```diff
@@ -99,50 +100,63 @@ int spi_config(const struct device *dev, uint32_t frequency,
 	return 0;
 }
 
+static ALWAYS_INLINE bool spi_sifive_send_available(const struct device *dev) {
+	return !(sys_read32(SPI_REG(dev, REG_TXDATA)) & SF_TXDATA_FULL);
+}
+
+static ALWAYS_INLINE
 void spi_sifive_send(const struct device *dev, uint16_t frame)
 {
-	while (sys_read32(SPI_REG(dev, REG_TXDATA)) & SF_TXDATA_FULL) {
-	}
-
 	sys_write32((uint32_t) frame, SPI_REG(dev, REG_TXDATA));
 }
 
-uint16_t spi_sifive_recv(const struct device *dev)
+static ALWAYS_INLINE
+bool spi_sifive_recv(const struct device *dev, uint16_t *val)
 {
-	uint32_t val;
+	uint32_t reg = sys_read32(SPI_REG(dev, REG_RXDATA));
 
-	while ((val = sys_read32(SPI_REG(dev, REG_RXDATA))) & SF_RXDATA_EMPTY) {
+	if (reg & SF_RXDATA_EMPTY) {
+		return false;
 	}
-
-	return (uint16_t) val;
+	*val = reg;
```

Comment:
It's better to keep this explicit conversion from `uint32_t` to `uint16_t`, otherwise compilers may issue a warning for this, depending on the configuration used (see `-Wconversion` option in gcc, for example).

Reply:
Fixed.